Add API for YandexGPT model

VycheslavRussu opened the pull request:
Файл с API до Yandex GPT, который позволяет отправлять запросы на модель и получать ответы от нее. Хранит историю запросов внутри себя.

VycheslavRussu committed:
Added class YandexGPT

VycheslavRussu committed:
Added class ContextStorage

VycheslavRussu committed:
Added class UseCase

VycheslavRussu committed:
Added class UseCase

VycheslavRussu committed:
First version is ready

sedirov07 committed:
добавлены зависимости и гитигнор

sedirov07 committed:
Добавлено приложение на streamlit

sedirov07 committed:
Добавлена обработка текста docx, примерное использование моделей

sedirov07 committed:
добавлена зависимость для работы с docx форматом

sedirov07 committed:
Изменены зависимости и Streamlit чат с моделью, добавлена интеграция с yandex-gpt

sedirov07 commented on yandex-gpt-api.py:
```diff
@@ -0,0 +1,214 @@
+        }
+    ]
+
+    def setup_context(self):
```
Вместо метода setup_context добавить в методе __init__ вызов self.context_storage.set_default_context(self.setup_messages)

VycheslavRussu replied:
Пофиксил

sedirov07 commented on yandex-gpt-api.py:
```diff
@@ -0,0 +1,214 @@
+    """
+
+    def __init__(self, folder_id, api_key):
+        self.__FOLDER_ID = folder_id
```
Убрать передачу параметров в метод __init__

VycheslavRussu replied:
Исправил

sedirov07 reviewed (changes requested):
Внести изменения, указанные в комментариях к файлу yandex-gpt-api.py построчно

sedirov07 committed:
добавлена интеграция с yandex_ocr_api

VycheslavRussu committed:
Merge remote-tracking branch 'origin/streamlit-app' into yandex-gpt-api

VycheslavRussu committed:
Updates:
1. File name has been changed
2. Location has been fixed
3. Secrets were updated
4. Class UseCase has been simplified

VycheslavRussu committed:
Merge branch 'main' into yandex-gpt-api

VycheslavRussu committed:
Updates:
1. Secrets were updated

VycheslavRussu committed:
Merge remote-tracking branch 'origin/yandex-gpt-api' into yandex-gpt-api

VycheslavRussu committed:
Updates:
1. Requirements were updated

Airmareing commented on models_api/yandex_gpt/yandex_gpt_api.py:
```diff
@@ -0,0 +1,183 @@
+        return reply_message["text"]
+
+
+# # Создали экземпляр класса
```
Необходимо оформить пример как unit-тест в отдельном файле

VycheslavRussu replied:
Ок, вынесу и добавлю в файл с тестами, когда создадим его

VycheslavRussu replied:
Сейчас просто уберу из кода

Airmareing reviewed (changes requested):
см комментарий об оформлении примера

VycheslavRussu committed:
Updates:
1. Test example wes deleted
